python-package/lightgbm/basic.py

```diff
@@ -248,10 +249,11 @@ def register_logger(
     if not _has_method(logger, info_method_name) or not _has_method(logger, warning_method_name):
         raise TypeError(f"Logger must provide '{info_method_name}' and '{warning_method_name}' method")
 
-    global _LOGGER, _INFO_METHOD_NAME, _WARNING_METHOD_NAME
+    global _LOGGER, _INFO_METHOD_NAME, _WARNING_METHOD_NAME, _LOG_KWARGS
     _LOGGER = logger
     _INFO_METHOD_NAME = info_method_name
     _WARNING_METHOD_NAME = warning_method_name
+    _LOG_KWARGS = {"stacklevel": 2} if isinstance(logger, logging.Logger) else {}
```

Comment:
please make this an `if-else` instead of a 1-liner. We have a slight preference for that, for readability.